ADAL/src/cache/ios/ADKeychainTokenCache.m

```diff
@@ -298,14 +298,22 @@ - (BOOL)removeAllForClientId:(NSString * __nonnull)clientId
     
     BOOL deleteSuccessful = YES;
     NSArray* items = [self allItems:nil];
+    
+    ADAuthenticationError* adError = nil;
+    
     for (ADTokenCacheItem * item in items)
     {
         if ([clientId isEqualToString:[item clientId] ])
         {
-            [self removeItem:item error:error];
-            if (*error)
+            [self removeItem:item error:&adError];
+            if (adError)
             {
                 deleteSuccessful = NO;
+                
+                if (error)
```

Comment:
Question: why doesn't the __nullable keyword guarantee *if (error)* being true here? :P